packages/lesswrong/components/posts/PostsPage/LWPostsPageHeader.tsx

```diff
@@ -0,0 +1,219 @@
+      maxWidth: "calc(100% - 60px)",
+      fontSize: theme.typography.body2.fontSize,
+    },
+  },
+  feedName: {
+    fontSize: theme.typography.body2.fontSize,
+    [theme.breakpoints.down('sm')]: {
+      display: "none"
+    }
+  },
+  topRight: {
+    position: 'absolute',
+    right: 8, 
+    top: -48,
+    display: 'flex',
+    [theme.breakpoints.down('sm')]: {
+      top: -16,
+      marginTop: 8,
+      marginBottom: 8
+    },
+    [theme.breakpoints.down('xs')]: {
+      display: 'none'
+    }
+  },
+  audioPlayerWrapper: {
+    position: 'absolute',
+    right: 8, 
+    top: 15,
+    display: 'flex',
+    [theme.breakpoints.down('sm')]: {
+      top: 8,
+      right: 8
+    }
+  },
+  sequenceNav: {
+    marginBottom: 8,
+    marginTop: -22
+  },
+  eventData: {
+    marginTop: 48
+  },
+  titleSection: {
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'space-between'
+  },
+  title: {
+    [theme.breakpoints.down('sm')]: {
+      marginBottom: 10
+    }
+  },
+  mobileHeaderVote: {
+    textAlign: 'center',
+    fontSize: 42,
+    [theme.breakpoints.up("sm")]: {
+      display: 'none'
+    }
+  },
+  date: {
+    marginTop: 8,
+    marginBottom: 8
+  },
+  mobileButtons: {
+    display: 'flex',
+    alignItems: 'center',
+    [theme.breakpoints.up('sm')]: {
+      display: 'none'
+    }
+  }
+}); 
+
+// On the server, use the 'url' library for parsing hostname out of feed URLs.
+// On the client, we instead create an <a> tag, set its href, and extract
+// properties from that. (There is a URL class which theoretically would work,
+// but it doesn't have the hostname field on IE11 and it's missing entirely on
+// Opera Mini.)
+const URLClass = getUrlClass()
+
+function getProtocol(url: string): string {
+  if (isServer)
+    return new URLClass(url).protocol;
+
+  // From https://stackoverflow.com/questions/736513/how-do-i-parse-a-url-into-hostname-and-path-in-javascript
+  var parser = document.createElement('a');
+  parser.href = url;
+  return parser.protocol;
+}
+
+function getHostname(url: string): string {
+  if (isServer)
+    return new URLClass(url).hostname;
+
+  // From https://stackoverflow.com/questions/736513/how-do-i-parse-a-url-into-hostname-and-path-in-javascript
+  var parser = document.createElement('a');
+  parser.href = url;
+  return parser.hostname;
+}
+
+
+/// LWPostsPageHeader: The metadata block at the top of a post page, with
+/// title, author, voting, an actions menu, etc.
+const LWPostsPageHeader = ({post, answers = [], dialogueResponses = [], showEmbeddedPlayer, toggleEmbeddedPlayer, hideMenu, hideTags, annualReviewMarketInfo, classes}: {
+  post: PostsWithNavigation|PostsWithNavigationAndRevision|PostsListWithVotes,
+  answers?: CommentsList[],
+  dialogueResponses?: CommentsList[],
+  showEmbeddedPlayer?: boolean,
+  toggleEmbeddedPlayer?: () => void,
+  hideMenu?: boolean,
+  hideTags?: boolean,
+  annualReviewMarketInfo?: AnnualReviewMarketInfo,
+  classes: ClassesType,
```

Comment:
missing `<typeof styles>`, and `styles` needs the return type annotation removed